resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -361,13 +360,12 @@ private boolean handleRedirect(
                 }
 
                 if (nameServers.isEmpty()) {
-                    promise.tryFailure(
-                            new UnknownHostException("Unable to find correct name server for " + hostname));
+                    addTrace(envelope.sender(), "no matching authoritative name server found");
```

Comment:
consider adding a bit more clarity to this trace statement: `"no matching authoritative name server found"` -> `"no matching authoritative name server found in the ADDITIONALS section."`